src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -305,5 +320,19 @@ private static AgentComponents BuildAgentComponents(string dbgInstanceName)
 
 			return agentComponents;
 		}
+
+		private static void SafeAgentSetup(AgentComponents agentComponents)
+		{
+			try
+			{
+				Agent.Setup(agentComponents);
+			}
+			catch (Agent.InstanceAlreadyCreatedException ex)
+			{
+				Agent.Instance.Logger.Error()?.LogException(ex, "Agent's singleton instance already exists before call to"
```

Comment:
I'm not sure users would understand this error message. Would maybe something like this easier to understand for someone who is not familiar with the agent's code? 

```
The Elastic APM agent was already initialized before call to {nameof(ElasticApmModule)}.{nameof(Init)} - {nameof(ElasticApmModule)} will use existing instance
```

Also, I'm not sure about the `even though it might lead to using incorrect configuration...` part. Maybe I'd just say 

```
This could lead to unexpected behaviour.
```

Reply:
How about?

>  This might lead to unexpected behavior (for example agent using incorrect configuration source such as environment variables instead of Web.config).

Reply:
That sounds good to me. 

Would we also change the 1. part? My thinking behind that is that explicitly saying "agent was already initialized" can be easier to understand than "Agent's singleton instance already exists" because that would make them wonder about what singleton it is and how they could change it.

Reply:
Agreed.